### PR TITLE
JAMES-2055  Add an option to desactivate CONDSTORE and make it false by default

### PR DESCRIPTION
**Pull request by aduprat**
Only 4 last commits:

JAMES-2055 Remove static methods 			ef6bb31
JAMES-2055 Add ability to manage CONDSTORE in IMAP configuration 			dfa7a76
JAMES-2055 Activate or not CONDSTORE in capabilities 			cb710c6
JAMES-2055 Add MPT test on configurable capabilities 			41d54c7

### CODE REVIEW

File: protocols/imap/src/main/java/org/apache/james/imap/api/ImapConfiguration.java [hunk 1]
```diff
@@ -59,16 +59,20 @@ private static boolean noBlankString(String disableCap) {
             return !StringUtils.isBlank(disableCap);
         }
 
+        private static final boolean DEFAULT_IS_CONDSTORE_UNABLE = false;
+
         private Optional<Long> idleTimeInterval;
         private Optional<TimeUnit> idleTimeIntervalUnit;
         private Optional<Boolean> enableIdle;
         private ImmutableSet<String> disabledCaps;
+        private Optional<Boolean> isCondstoreUnable;
```
Comment (chibenwa) [inline protocols/imap/src/main/java/org/apache/james/imap/api/ImapConfiguration.java]: s/Unable/enable/g

File: protocols/imap/src/main/java/org/apache/james/imap/api/ImapConfiguration.java [hunk 2]
```diff
@@ -59,16 +59,20 @@ private static boolean noBlankString(String disableCap) {
             return !StringUtils.isBlank(disableCap);
         }
 
+        private static final boolean DEFAULT_IS_CONDSTORE_UNABLE = false;
```
Comment (chibenwa) [inline protocols/imap/src/main/java/org/apache/james/imap/api/ImapConfiguration.java]: DEFAULT_CONDSTORE_DISABLED ?

File: mpt/core/src/main/java/org/apache/james/mpt/api/ImapHostSystem.java [hunk 3]
```diff
@@ -29,4 +30,5 @@
 
     void setQuotaLimits(long maxMessageQuota, long maxStorageQuota) throws Exception;
 
+    void configure(ImapConfiguration imapConfiguration);
```
Comment (chibenwa) [inline mpt/core/src/main/java/org/apache/james/mpt/api/ImapHostSystem.java]: IMO it should be an available method only for JamesHostSystem...

File: mpt/impl/imap-mailbox/core/src/main/java/org/apache/james/mpt/host/JamesImapHostSystem.java [hunk 4]
```diff
@@ -159,5 +161,21 @@ private HierarchicalConfiguration userRepositoryConfiguration() {
         configuration.addProperty("administratorId", "imapuser");
         return configuration;
     }
-    
+
+    @Override
+    public boolean supports(Feature... features) {
+        // TODO Auto-generated method stub
```
Comment (chibenwa) [inline mpt/impl/imap-mailbox/core/src/main/java/org/apache/james/mpt/host/JamesImapHostSystem.java]: Remove TODO comments?

File: mpt/impl/imap-mailbox/inmemory/src/test/java/org/apache/james/mpt/imapmailbox/inmemory/Condstore.java [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+public class Condstore extends BaseAuthenticatedState {
+
+    @Inject
+    private static ImapHostSystem system;
+    
+    public Condstore() throws Exception {
+        super(system);
+    }
+
+    @Test
```
Comment (chibenwa) [inline mpt/impl/imap-mailbox/inmemory/src/test/java/org/apache/james/mpt/imapmailbox/inmemory/Condstore.java]: You should also have a test when you set CONDSTORE explicily to false.

**Comment by chibenwa** [comment]
```
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536] condstoreShouldBeDisableByDefault(org.apache.james.mpt.imapmailbox.inmemory.Condstore)  Time elapsed: 0.003 sec  <<< ERROR!
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536] java.lang.NullPointerException
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536] 
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536] condstoreShouldBeDisableByDefault(org.apache.james.mpt.imapmailbox.inmemory.Condstore)  Time elapsed: 0.003 sec  <<< ERROR!
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536] java.lang.NullPointerException
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536] 
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536] condstoreShouldBeEnableWhenGivenAndTrue(org.apache.james.mpt.imapmailbox.inmemory.Condstore)  Time elapsed: 0 sec  <<< ERROR!
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536] java.lang.NullPointerException
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536] 
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536] condstoreShouldBeEnableWhenGivenAndTrue(org.apache.james.mpt.imapmailbox.inmemory.Condstore)  Time elapsed: 0 sec  <<< ERROR!
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536] java.lang.NullPointerException
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536] 
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536] 
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536] Results :
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536] 
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536] Tests in error: 
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536] org.apache.james.mpt.imapmailbox.inmemory.Condstore.condstoreShouldBeDisableByDefault(org.apache.james.mpt.imapmailbox.inmemory.Condstore)
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536]   Run 1: PASS
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536]   Run 2: Condstore>BaseAuthenticatedState.setUp:75->BaseImapProtocol.setUp:37->AbstractProtocolTestFramework.setUp:63 » NullPointer
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536]   Run 3: Condstore>AbstractSimpleScriptedTestProtocol.tearDown:64->AbstractProtocolTestFramework.tearDown:68 » NullPointer
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536] 
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536] org.apache.james.mpt.imapmailbox.inmemory.Condstore.condstoreShouldBeEnableWhenGivenAndTrue(org.apache.james.mpt.imapmailbox.inmemory.Condstore)
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536]   Run 1: PASS
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536]   Run 2: Condstore>BaseAuthenticatedState.setUp:75->BaseImapProtocol.setUp:37->AbstractProtocolTestFramework.setUp:63 » NullPointer
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536]   Run 3: Condstore>AbstractSimpleScriptedTestProtocol.tearDown:64->AbstractProtocolTestFramework.tearDown:68 » NullPointer
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536] 
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536] 
[Building Java 8 james-2055-29fcf952-c21f-4224-963a-04cf1edfa536] Tests run: 613, Failures: 0, Errors: 2, Skipped: 0
```

File: protocols/imap/src/test/java/org/apache/james/imap/processor/CapabilityProcessorTest.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+    @Test
+    public void condstoreShouldBeSupportedWhenSelectedFor() {
+        testee.configure(ImapConfiguration.builder().isCondstoreUnable(true).build());
+
+        Set<String> supportedCapabilities = testee.getSupportedCapabilities(null);
+        assertThat(supportedCapabilities).contains(ImapConstants.SUPPORTS_CONDSTORE);
+    }
+
+    @Test
+    public void condstoreShouldBeNotSupportedBeDefault() {
```
Comment (quynhn) [inline protocols/imap/src/test/java/org/apache/james/imap/processor/CapabilityProcessorTest.java]: s/BeDefault/ByDefault/?

**Comment by aduprat** [comment]
Replaced by #841

**Pull request closed without merging**